### ISSUE TITLE
zoxide script crashes at `z<space><char>`

### ISSUE THREAD

**Issue by fdncred** [nushell/engine-q#612]
I'm working on this zoxide script to get it working with engine-q. I believe `shells` doesn't exist yet but I can still `source` the file and when I try to do `z<space<char>` it crashes like this.

```
〉z thread 'main' panicked at 'internal error: span missing in file contents cache', crates/nu-protocol/src/engine/engine_state.rs:380:9
note: run with RUST_BACKTRACE=1 environment variable to display a backtrace
```
here's the script.
```
# Code generated by zoxide. DO NOT EDIT.

# =============================================================================
#
# Utility functions for zoxide.
#

# Default prompt for Nushell.
def __zoxide_prompt [] {
    # if gstat plugin is installed could do `gstat | get branch`
    let git = $'(do -i {git rev-parse --abbrev-ref HEAD} | decode utf-8 | str trim -r -c (char nl))'
    let git = (if ($git | str length) == 0 { '' } else {
        build-string (char lp) (ansi cb) $git (ansi reset) (char rp)
    })
    build-string (ansi gb) ($nu.cwd) (ansi reset) $git '> '
}

# =============================================================================
#
# Hook configuration for zoxide.
#

## Darren's shells HACK
def shells [] {
    [[active name path]; [$true filesystem $nu.cwd]]
}
## end HACK

# Hook to add new entries to the database.
def __zoxide_hook [] {
    shells | where active == $true && name == filesystem | get path | each {
        zoxide add -- $it
    }
}

# Initialize hook.
let-env PROMPT_COMMAND = (
    # let prompt = (if (env | where name == PROMPT_COMMAND | length) == 0 {
    let prompt = (if ("PROMPT_COMMAND" not-in (env).name)  {
        "__zoxide_prompt"
    } else { 
        $nu.env.PROMPT_COMMAND
    });

    if ($prompt | str contains '__zoxide_hook') { $prompt } else { $'__zoxide_hook;($prompt)' }
)

# =============================================================================
#
# When using zoxide with --no-aliases, alias these internal functions as
# desired.
#

# Jump to a directory using only keywords.
def __zoxide_z [...rest:string] {
    if (shells | where active == $true | get name).0 != filesystem {
        if ($rest | length) > 1 {
            $'zoxide: can only jump directories on filesystem(char nl)'
        } else {
            cd $rest
        }
    } else {
        let arg0 = ($rest | append '~' | first 1);
        if ($rest | length) <= 1 && ($arg0 == '-' || ($arg0 | path expand | path exists)) {
            cd $arg0
        } else {
            cd $'(zoxide query --exclude ($nu.cwd) -- $rest | decode utf-8 | str trim -r -c (char nl))'
        }
    }
}

# Jump to a directory using interactive search.
def __zoxide_zi  [...rest:string] {
    if (shells | where active == $true | get name).0 != filesystem {
        $'zoxide: can only jump directories on filesystem(char nl)'
    } else {
        cd $'(zoxide query -i -- $rest | decode utf-8 | str trim -r c (char nl))'
    }
}

# =============================================================================
#
# Convenient aliases for zoxide. Disable these using --no-aliases.
#

alias z = __zoxide_z
alias zi = __zoxide_zi

# =============================================================================
#
# To initialize zoxide, add this to your configuration (find it by running
# `config path` in Nushell):
#
#   startup = ['zoxide init nushell --hook prompt | save ~/.zoxide.nu', 'source ~/.zoxide.nu']
#
# Note: zoxide only supports Nushell v0.37.0 and above.
```
updated script 12/28

**Comment by fdncred** [comment]
I've started to narrow this down. the custom command `__zoxide_z <string>` works great. however, if I call the alias `z <string>` the first character of `<string>` causes this panic.

```
z thread 'main' panicked at 'internal error: span missing in file contents cache', crates/nu-protocol/src/engine/engine_state.rs:380:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
backtrace
```
z thread 'main' panicked at 'internal error: span missing in file contents cache', crates/nu-protocol/src/engine/engine_state.rs:380:9
stack backtrace:
   0: rust_begin_unwind
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/std/src/panicking.rs:517:5
   1: core::panicking::panic_fmt
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/core/src/panicking.rs:100:14
   2: core::panicking::panic
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/core/src/panicking.rs:50:5
   3: nu_protocol::engine::engine_state::EngineState::get_span_contents
             at ./engine-q/crates/nu-protocol/src/engine/engine_state.rs:380:9
   4: nu_protocol::engine::engine_state::StateWorkingSet::get_span_contents
             at ./engine-q/crates/nu-protocol/src/engine/engine_state.rs:833:20
   5: nu_parser::parser::parse_call
             at ./engine-q/crates/nu-parser/src/parser.rs:684:20
   6: nu_parser::parser::parse_expression
             at ./engine-q/crates/nu-parser/src/parser.rs:3162:9
   7: nu_parser::parser::parse_call
             at ./engine-q/crates/nu-parser/src/parser.rs:697:37
   8: nu_parser::parser::parse_expression
             at ./engine-q/crates/nu-parser/src/parser.rs:3162:9
   9: nu_parser::parser::parse_statement
             at ./engine-q/crates/nu-parser/src/parser.rs:3262:31
  10: nu_parser::parser::parse_block::{{closure}}
             at ./engine-q/crates/nu-parser/src/parser.rs:3405:35
  11: core::iter::adapters::map::map_fold::{{closure}}
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/core/src/iter/adapters/map.rs:84:28
  12: core::iter::traits::iterator::Iterator::fold
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/core/src/iter/traits/iterator.rs:2171:21
  13: <core::iter::adapters::map::Map<I,F> as core::iter::traits::iterator::Iterator>::fold
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/core/src/iter/adapters/map.rs:124:9
  14: core::iter::traits::iterator::Iterator::for_each
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/core/src/iter/traits/iterator.rs:737:9
  15: <alloc::vec::Vec<T,A> as alloc::vec::spec_extend::SpecExtend<T,I>>::spec_extend
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/alloc/src/vec/spec_extend.rs:40:17
  16: <alloc::vec::Vec<T> as alloc::vec::spec_from_iter_nested::SpecFromIterNested<T,I>>::from_iter
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/alloc/src/vec/spec_from_iter_nested.rs:56:9
  17: <alloc::vec::Vec<T> as alloc::vec::spec_from_iter::SpecFromIter<T,I>>::from_iter
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/alloc/src/vec/spec_from_iter.rs:33:9
  18: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/alloc/src/vec/mod.rs:2517:9
  19: core::iter::traits::iterator::Iterator::collect
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/core/src/iter/traits/iterator.rs:1745:9
  20: <nu_protocol::ast::block::Block as core::convert::From<T>>::from
             at ./engine-q/crates/nu-protocol/src/ast/block.rs:61:20
  21: <T as core::convert::Into<U>>::into
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/core/src/convert/mod.rs:542:9
  22: nu_parser::parser::parse_block
             at ./engine-q/crates/nu-parser/src/parser.rs:3366:24
  23: nu_parser::parser::parse
             at ./engine-q/crates/nu-parser/src/parser.rs:3684:25
  24: <nu_cli::syntax_highlight::NuHighlighter as reedline::highlighter::Highlighter>::highlight
             at ./engine-q/crates/nu-cli/src/syntax_highlight.rs:17:30
  25: reedline::engine::Reedline::prepare_buffer_content
             at /Users/fdncred/.cargo/git/checkouts/reedline-e42026a78d91c510/afa83d9/src/engine.rs:951:32
  26: reedline::engine::Reedline::buffer_paint
             at /Users/fdncred/.cargo/git/checkouts/reedline-e42026a78d91c510/afa83d9/src/engine.rs:979:40
  27: reedline::engine::Reedline::repaint
             at /Users/fdncred/.cargo/git/checkouts/reedline-e42026a78d91c510/afa83d9/src/engine.rs:899:13
  28: reedline::engine::Reedline::run_edit_commands
             at /Users/fdncred/.cargo/git/checkouts/reedline-e42026a78d91c510/afa83d9/src/engine.rs:814:21
  29: reedline::engine::Reedline::handle_editor_event
             at /Users/fdncred/.cargo/git/checkouts/reedline-e42026a78d91c510/afa83d9/src/engine.rs:565:17
  30: reedline::engine::Reedline::handle_event
             at /Users/fdncred/.cargo/git/checkouts/reedline-e42026a78d91c510/afa83d9/src/engine.rs:427:13
  31: reedline::engine::Reedline::read_line_helper
             at /Users/fdncred/.cargo/git/checkouts/reedline-e42026a78d91c510/afa83d9/src/engine.rs:403:39
  32: reedline::engine::Reedline::read_line
             at /Users/fdncred/.cargo/git/checkouts/reedline-e42026a78d91c510/afa83d9/src/engine.rs:313:22
  33: engine_q::main
             at ./engine-q/src/main.rs:410:25
  34: core::ops::function::FnOnce::call_once
             at /rustc/f1edd0429582dd29cccacaf50fd134b05593bd9c/library/core/src/ops/function.rs:227:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Comment by fdncred** [comment]
fixed by #713
